src/node_buffer.cc

```diff
@@ -724,6 +724,8 @@ void StringWrite(const FunctionCallbackInfo<Value>& args) {
                                           &max_length));
 
   max_length = std::min(ts_obj_length - offset, max_length);
+  if (max_length > static_cast<size_t>(v8::String::kMaxLength))
+    ThrowErrStringTooLong(env->isolate());
```

Comment:
```suggestion
  if (max_length > static_cast<size_t>(v8::String::kMaxLength)) {
    ThrowErrStringTooLong(env->isolate());
    return;
  }
```

You're missing a return here; the Throw* functions don't actually return, you have to do it manually.